lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -239,6 +242,21 @@ def add_metadata(doc, options)
         end
       end
 
+      def add_transaction_meta_data(doc, options = {})
+        transaction_meta_data = options[:transaction_meta_data] || []
+        return if transaction_meta_data.empty?
+
+        doc.send('transaction-meta-data') do
+          options[:transaction_meta_data].each do |meta|
+            doc.send('meta-data') do
+              doc.send('meta-key', meta[:key])
+              doc.send('meta-value', meta[:value])
+              doc.send('meta-description', meta[:description])
```

Comment:
Based on the documentation I'm seeing [here](https://developers.bluesnap.com/v8976-XML/docs/meta-data), there is an `is-visible` boolean that can also be added to the `meta-data` object.

I'm wondering why you chose to include `transaction-meta-data` in the refund request at all? The work ticket does not mention it, and it appears from the gateway documentation that it is entirely optional. I also appears that it can be added to any transaction type, not just refund, so I'm wondering if we do decide to add it, shouldn't we add it to every transaction?

Reply:
meta data fields are already being sent in every transaction using method [add_meta_data](https://github.com/activemerchant/active_merchant/blob/ff776e257d3c605ec76c7f9f2620b7cc9a8e5098/lib/active_merchant/billing/gateways/blue_snap.rb#L216) in the code. So this function is reused for refund as well (previously added method is removed)

Reply:
Nice, I didn't notice `add_meta_data` was already available. Good change 👍